tools/clang/unittests/SPIRV/CodeGenSPIRVTest.cpp

```diff
@@ -270,6 +270,9 @@ TEST_F(FileTest, CastImplicit2UInt) { runFileTest("cast.2uint.implicit.hlsl"); }
 TEST_F(FileTest, CastExplicit2UInt) { runFileTest("cast.2uint.explicit.hlsl"); }
 TEST_F(FileTest, CastImplicit2FP) { runFileTest("cast.2fp.implicit.hlsl"); }
 TEST_F(FileTest, CastExplicit2FP) { runFileTest("cast.2fp.explicit.hlsl"); }
+TEST_F(FileTest, CastImplicitFlatConversion) {
+  runFileTest("cast.flat-conversion.implicit.hlsl");
```

Comment:
This is actually explicit cast?

Reply:
FlatConversion is the CastKind for an ImplicitCastExpr. That's why I chose implicit.

Reply:
Hmm... Actually we are casting explicitly in the code: `... = (SomeStruct)0`. I think this is explicit cast from C languague's perspective. Implicit cast means something like `float f = <some-int>`, which is done implicitly.

Anyway, doesn't matter that much.